### ISSUE TITLE
The plugin is loading too much RAM and increases CPU consumption.

### ISSUE THREAD

**Issue by fxss-repository** [J-Tanzanite/Lilac-SourceTV#1]
The plugin loads about 400 MB of something into RAM during server boot.
Memory consumption by the end of the day is about 1.3GB. Without your plugin, about 800MB.
It was also noticed that the consumption of CPU time increased by 13% to nowhere.

**Comment by J-Tanzanite** [comment]
That doesn't sound right.
There's nothing in the code that takes up that much memory, the binary itself is 7.5KB, which is rather puny.

The only thing that could possible take up more RAM is the timers "timer_decrement_aim...bot/lock".
However, I highly doubt that timers take up that much ram usage... If that were the case, then Sourcemod would have big issues as timers are used constantly and everywhere (And this plugin uses them sparingly).

I did notice I forgot to use the flags "TIMER_FLAG_NO_MAPCHANGE", but that wouldn't make THAT big of a difference.

How did you go about measuring RAM? And also CPU consumption? And also, what OS?

**Comment by WOLFA22** [comment]
OS - Windows NT 7 (7601)Meta - 971 1.10.7devSourcemod - 1.10.6492Measured by constantly restarting the server with your plugin and without your plugin, I watched the load through the program Process Hacker 3. Usually, the server process consumes about 436 MB without this plugin, with your plugin it consumes about 800 MB. Moreover, the server has loaded, and it loads something else with your plugin. What specifically I did not find out, but I can look in more detail in the coming days. CPU time is wasted after a while. (About 2 hours)10.08.2020, 20:17, "J_Tanzanite" <notifications@github.com>:
That doesn't sound right.
There's nothing in the code that takes up that much memory, the binary itself is 7.5KB, which is rather puny.
The only thing that could possible take up more RAM is the timers "timer_decrement_aim...bot/lock".
However, I highly doubt that timers take up that much ram usage... If that were the case, then Sourcemod would have big issues as timers are used constantly and everywhere (And this plugin uses them sparingly).
I did notice I forgot to use the flags "TIMER_FLAG_NO_MAPCHANGE", but that wouldn't make THAT big of a difference.
How did you go about measuring RAM? And also CPU consumption? And also, what OS?

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.
-- TypX+7 (983) 356-81-94gbelodedov2@yandex.ru
